pkg/externalLink/ExternalLinkService.go

```diff
@@ -93,252 +149,392 @@ func (impl ExternalLinkServiceImpl) Create(requests []*ExternalLinkDto, userId i
 				InternalMessage: "external link failed to create in db",
 				UserMessage:     "external link failed to create in db",
 			}
-			return nil, err
+			return externalLinksCreateUpdateResponse, err
 		}
-
-		for _, clusterId := range request.ClusterIds {
-			externalLinkClusterMapping := &ExternalLinkClusterMapping{
+		//for all identifiers, check if it is clusterLevel/appLevel
+		//if appLevel, get type and identifier else get clusterId
+		//save it in external_link_type_mapping table
+		linkType := request.Type
+		if linkType == APP_LEVEL_LINK && len(request.Identifiers) == 0 {
+			identifier := LinkIdentifier{
+				Type: getType(APP),
+			}
+			request.Identifiers = append(request.Identifiers, identifier)
+		}
+		for _, linkIdentifier := range request.Identifiers {
+			if linkType == CLUSTER_LEVEL_LINK {
+				linkIdentifier.Type = getType(CLUSTER)
+				linkIdentifier.Identifier = ""
+			} else if linkType == APP_LEVEL_LINK {
+				if linkIdentifier.Type == getType(DEVTRON_APP) || linkIdentifier.Type == getType(DEVTRON_INSTALLED_APP) {
+					appId, err := strconv.Atoi(linkIdentifier.Identifier)
```

Comment:
make a function to extract out appId from identifier, have been used at multiple places